lib/workers/bin/run_single_worker.rb

```diff
@@ -57,14 +57,22 @@
 worker_class = worker_class.constantize
 worker_class.before_fork
 unless options[:dry_run]
+  create_options = {}
+  runner_options = {}
+
+  if ENV["QUEUE"]
+    create_options[:queue_name] = ENV["QUEUE"]
+    runner_options[:ems_id] = worker_class.ems_id_from_queue_name(ENV["QUEUE"]) if worker_class.respond_to?(:ems_id_from_queue_name)
+  end
+
```

Comment:
Doesn't have to happen in this PR, and am mostly sharing thoughts here, but with how complex this script is getting, I am wondering if we should move the bulk of this code out of this script and make it testable.  Again, probably something for another PR.